exporter/datadogexporter/internal/logs/sender.go

```diff
@@ -26,11 +26,14 @@ import (
 	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter/internal/clientutil"
 )
 
+type handleSubmitLogFn func(ctx context.Context, s *Sender, batch []datadogV2.HTTPLogItem, tags string) error
+
 // Sender submits logs to Datadog intake
 type Sender struct {
-	logger  *zap.Logger
-	api     *datadogV2.LogsApi
-	verbose bool // reports whether payload contents should be dumped when logging at debug level
+	logger          *zap.Logger
+	api             *datadogV2.LogsApi
+	verbose         bool // reports whether payload contents should be dumped when logging at debug level
+	handleSubmitLog handleSubmitLogFn
```

Comment:
No need for this hackery. You can use `httptest.Server` and assert _exactly_ what's being flushed by `datadogV2.LogsApi`. Simply create a new `Sender` struct with an `api` field that [uses the test server's endpoint](https://github.com/liustanley/opentelemetry-collector-contrib/blob/a8e94f69abbd690b5cd2047523561a2f9ab056a8/exporter/datadogexporter/internal/logs/sender.go#L50).

Reply:
Whenever you are adding logic into code that is _specifically_ meant for simplifying tests, always question it. Try to never sacrifice architecture for the ease of tests, but rather view the difficulty of adding tests as a question with regards to the quality of the architecture. That's the general view IMHO. It doesn't apply here, since the architecture is fine and you should be able to test with the previous comment's suggestion.

Reply:
We can use https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/ad5e5bbc42b3095ef34de9f71df472a5ac21325b/exporter/datadogexporter/internal/testutil/logs.go#L26 that does this functionality

Reply:
Just updated this to use `DatadogLogsServer` and I added a new function in `logs.go` to use for testing each endpoint call